source/js/components/donation-amount-toggle.js

```diff
@@ -5,14 +5,33 @@ class AmountToggle {
 
   constructor(node) {
     this.toggleButton = node;
+    this.updateButton = document.querySelector("#js-update-button");
     this.actionsContainer = document.querySelector("[data-amount-actions]");
+    this.hiddenInput = document.querySelector("#js-donation-value");
     this.activeClass = "active";
+    this.selectedClass = "selected";
 
     this.bindEvents();
   }
 
+  UpdateHiddenInputFromInput() {
+    let input = document.querySelector("#js-update-donation-value");
+    this.hiddenInput.value = input.value;
```

Comment:
From experience on sams I think we need to check that this input is valid - i.e., that it can be parsed as a number - before we update the value.

Reply:
I've added some validation